### ISSUE TITLE
How to obtain the merchant's own sku code filled in when creating a offer in the endpoint

### ISSUE THREAD

**Issue by yangcen6** [allegro/allegro-api#9601]
![20240806-210550](https://github.com/user-attachments/assets/9cb91fbc-c081-42af-b07f-cb6f51ebbff9)
As shown in the yellow circle in the picture, when creating a offer in seller center, we will fill in our internal sku code in this place for shipment from our own warehouse. So in these two endpoints（https://api.{environment}/order/checkout-forms and https://api.{environment}/order/checkout-forms/{id}））, is the value of the external.id the content filled in the yellow box??

**Comment by Lukasz-Zurek** [comment]
No, it is a Manufacturer's code parameter (224017) which we do not return in the checkout-forms resources. You can pass an external.id while listing an offer via [POST /sale/product-offers](https://developer.allegro.pl/documentation/#operation/createProductOffers). Then you can use [GET /sale/offers/](https://developer.allegro.pl/documentation/#operation/searchOffersUsingGET) where we return this field, and you can also use a filter (external.id).

**Comment by yangcen6** [comment]
Can we only add this external.id through [POST /sale/product-offers]? Because we will not list offers through the endpoints, we will list offers in the seller center

**Comment by Lukasz-Zurek** [comment]
@yangcen6 Unfortunately you can't, because this resource is for listing an offer. Eventually you can edit an existing offer via [PATCH /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation/#operation/editProductOffers) which is for editing and where you can provide only **"external.id"** field. But first please check this resource closely. [Here](https://developer.allegro.pl/tutorials/how-to-process-list-of-offers-m09BKA5v8H3#editing-single-offer) is a tutorial.

**Comment by yangcen6** [comment]
That is, if we manually list an offer in the seller center, we can use this endpoint to add an external.id to the listed offer so that our own warehouse can ship the order with the external.id,right?

**Comment by Lukasz-Zurek** [comment]
Yes, you can add an external.id to the listed offer. We will return it in the orders too.

**Comment by yangcen6** [comment]
If we fill in the manufacturer code (our internal sku code) in the yellow box in the seller center when creating an offer, when the customer selects this offer and pays for the order, can we get this sku code though **/order/checkout-forms** or **/order/checkout-forms/{id}**？If can, which field?

**Comment by Lukasz-Zurek** [comment]
@yangcen6 No, we do not have this field in **/order/checkout-forms** or **/order/checkout-forms/{id}**.

**Comment by yangcen6** [comment]
So which endpoint can we use to get the content filled in this yellow box?

**Comment by Lukasz-Zurek** [comment]
As I have already mentioned before - we do not return this field in the orders. We only return it in the parameters for offers - if you have filled it. As many other parameters, it can be returned in the **"parameters"** section. More information [here](https://developer.allegro.pl/tutorials/how-to-process-list-of-offers-m09BKA5v8H3#list-of-offers).

**Comment by yangcen6** [comment]
![20240808-155525](https://github.com/user-attachments/assets/c541c31c-6f1c-4008-9a3e-43f114afb77e)
like the parameters of the endpoint /sale/product-offers/{offerId} ?

**Comment by Lukasz-Zurek** [comment]
Yes, exactly.
